tests/test_scheduler_plan_mode.rs

```diff
@@ -0,0 +1,307 @@
+                    working_directory_cleanup_config: WorkingDirectoryCleanupConfig::MaxExecutions(
+                        4,
+                    ),
+                    host: Host::Source,
+                    metadata: PlanMetadata {
+                        application: "app".into(),
+                        suite_name: "minimal_suite".into(),
+                        variant: "".into(),
+                    },
+                },
+                PlanConfig {
+                    id: "conda_plan".into(),
+                    source: Source::Manual {
+                        base_dir: suite_dir.into(),
+                    },
+                    robot_config: RobotConfig {
+                        robot_target: "tasks.robot".into(),
+                        top_level_suite_name: None,
+                        suites: vec![],
+                        tests: vec![],
+                        test_tags_include: vec![],
+                        test_tags_exclude: vec![],
+                        variables: vec![],
+                        variable_files: vec![],
+                        argument_files: vec![],
+                        exit_on_failure: false,
+                        environment_variables_rendered_obfuscated: vec![],
+                    },
+                    execution_config: ExecutionConfig {
+                        n_attempts_max: 1,
+                        retry_strategy: RetryStrategy::Complete,
+                        timeout: 10,
+                    },
+                    environment_config: EnvironmentConfig::Conda(CondaEnvironmentConfig {
+                        source: CondaEnvironmentSource::Manifest("conda.yaml".into()),
+                        robotmk_manifest_path: None,
+                        http_proxy_config: HTTPProxyConfig::default(),
+                        tls_certificate_validation: TlsCertificateValidation::Enabled,
+                        tls_revokation_enabled: true,
+                        build_timeout: 1200,
+                    }),
+                    session_config: SessionConfig::Current,
+                    working_directory_cleanup_config: WorkingDirectoryCleanupConfig::MaxExecutions(
+                        4,
+                    ),
+                    host: Host::Source,
+                    metadata: PlanMetadata {
+                        application: "app".into(),
+                        suite_name: "minimal_suite".into(),
+                        variant: "".into(),
+                    },
+                },
+            ],
+            execution_interval: 30,
+        }],
+    }
+}
+
+async fn run_scheduler(
+    test_dir: &Utf8Path,
+    config: &Config,
+    plan_id: Option<&str>,
+    no_plan_result: bool,
+    n_seconds_run_max: u64,
+) -> AnyhowResult<()> {
+    let config_path = test_dir.join("config.json");
+    write(&config_path, to_string(&config)?)?;
+    let run_flag_path = test_dir.join("run_flag");
```

Comment:
We can drop the run flag stuff here. In `test_scheduler`, we run in scheduling mode, so the scheduler should never terminate. Here, we should terminate after executing the plan once. So no need for the run flag, we can instead wait for the process to exit (successfully).